gitlab/resource_gitlab_user_impersonation_token.go

```diff
@@ -0,0 +1,129 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabUserImpersonationToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabUserImpersonationTokenCreate,
+		Read:   resourceGitlabUserImpersonationTokenRead,
+		Delete: resourceGitlabUserImpersonationTokenDelete,
+
```

Comment:
You should be able to also implement a pass-through importer, e.g.: https://github.com/claranet/terraform-provider-gitlab/blob/d10ba3ba63269eb0b3d9275f56e8fd7e471c5aed/gitlab/resource_gitlab_group.go#L22-L24